odl/solvers/advanced/proximal_operators.py

```diff
@@ -1079,6 +1080,91 @@ def _call(self, x, out):
     return ProximalCConjKL
 
 
+def proximal_cconj_kl_version_two(space, lam=1, g=None):
+    """Proximal operator factory of the convex conjugate of the KL divergence.
+
+    Function returning the proximal operator of the convex conjugate of the
+    functional F where F is the entropy-type Kullback-Leibler (KL) divergence
+
+        F(x) = sum_i (xln(pos(x)) - xln(g) + g - x)_i + ind_P(x)
+
+    with x and g in X and g non-negative. The indicator function ind_P(x)
+    for the positive elements of x is used to restrict the domain of F such
+    that F is defined over whole X. The non-negativity thresholding pos is
+    used to define F in the real numbers.
+
+    The convex conjugate F^* of F is
+
+        F^*(p) = sum_i (g (exp(p) - 1))_i
+
+    where p is the variable dual to x.
+
+    The proximal operator of the convex conjugate of F is
+
+        prox[sigma * (lam*F)^*](x)_i =
+            x_i - lam W(1/lam sigma g_i exp(1/lam * x_i))
+
+    where sigma is the step size-like parameter, lam the weighting in from of
+    the function F, and W is the Lambert W function (see, for example, the
+    `Wikipedia article <https://en.wikipedia.org/wiki/Lambert_W_function>`_).
+
+    Parameters
+    ----------
+    space : `DiscreteLp` or `ProductSpace` of `DiscreteLp` spaces
+        Space X which is the domain of the functional F
+    g : ``space`` element
+        Data term.
+    lam : positive float
+        Scaling factor.
+
+    Returns
+    -------
+    prox_factory : function
+        Factory for the proximal operator to be initialized.
+
+    Notes
+    -----
+    The functional is not well-defined without a prior g. Hence, if g is
+    omitted this will be interpreted as if g is equal to the one-element.
+
+    Write something about similarities and differences between the two
+    KL-versions?
+    """
+
+    # TODO: Update Notes-section in doc above.
+
+    lam = float(lam)
+
+    if g is not None and g not in space:
+        raise TypeError('{} is not an element of {}'.format(g, space))
+
+    class ProximalCConjKL(Operator):
```

Comment:
This name should not be the same as in the other implementation (copy-past error)